platform/kube/inject/inject.go

```diff
@@ -408,72 +503,29 @@ func IntoResourceFile(p *Params, in io.Reader, out io.Writer) error {
 		if err != nil {
 			return err
 		}
-		kinds := map[string]struct {
-			typ    interface{}
-			inject func(typ interface{}) error
-		}{
-			"Job": {
-				typ: &batchv1.Job{},
-				inject: func(typ interface{}) error {
-					o := typ.(*batchv1.Job)
-					return injectWithPolicyCheck(&o.ObjectMeta, &o.Spec.Template.ObjectMeta, &o.Spec.Template.Spec)
-				},
-			},
-			"DaemonSet": {
-				typ: &v1beta1.DaemonSet{},
-				inject: func(typ interface{}) error {
-					o := typ.(*v1beta1.DaemonSet)
-					return injectWithPolicyCheck(&o.ObjectMeta, &o.Spec.Template.ObjectMeta, &o.Spec.Template.Spec)
-				},
-			},
-			"ReplicaSet": {
-				typ: &v1beta1.ReplicaSet{},
-				inject: func(typ interface{}) error {
-					o := typ.(*v1beta1.ReplicaSet)
-					return injectWithPolicyCheck(&o.ObjectMeta, &o.Spec.Template.ObjectMeta, &o.Spec.Template.Spec)
-				},
-			},
-			"Deployment": {
-				typ: &v1beta1.Deployment{},
-				inject: func(typ interface{}) error {
-					o := typ.(*v1beta1.Deployment)
-					return injectWithPolicyCheck(&o.ObjectMeta, &o.Spec.Template.ObjectMeta, &o.Spec.Template.Spec)
-				},
-			},
-			"ReplicationController": {
-				typ: &v1.ReplicationController{},
-				inject: func(typ interface{}) error {
-					o := typ.(*v1.ReplicationController)
-					return injectWithPolicyCheck(&o.ObjectMeta, &o.Spec.Template.ObjectMeta, &o.Spec.Template.Spec)
-				},
-			},
-			"StatefulSet": {
-				typ: &appsv1beta1.StatefulSet{},
-				inject: func(typ interface{}) error {
-					o := typ.(*appsv1beta1.StatefulSet)
-					return injectWithPolicyCheck(&o.ObjectMeta, &o.Spec.Template.ObjectMeta, &o.Spec.Template.Spec)
-				},
-			},
```

Comment:
dumb question.. where are these being moved to?

Reply:
This code was generalized (see [lines 512-513](https://github.com/istio/pilot/pull/1104/files/dbdcebb82793fc91408bdba443d3eaf48a53d593#diff-3f71ca8a1c3836bc4a3360d83741f2fbR512) and [intoObject](https://github.com/istio/pilot/pull/1104/files#diff-3f71ca8a1c3836bc4a3360d83741f2fbR447)). This uses k8s `schema` and `runtime` support to manipulate k8s objects and the `reflect` package to extract the `ObjectMeta` and `PodTemplateSpec` sub-structures.

The table of supported GroupVersionKind's and corresponding objects is [here](https://github.com/istio/pilot/pull/1104/files#diff-bbaf00e98adfa4174f7d712a3869dff1R64).